services/net/filemonitor/FilemonitorAction.cs

```diff
@@ -323,21 +457,28 @@ private string GetFmsData(string story, string key, IngestModel ingest)
                 // Iterate over the list of stories and add a new item to the articles list for each story.
                 foreach (XmlElement story in elementList)
                 {
-                    var item = new SourceContent();
                     var papername = GetXmlData(story, Fields.Papername, ingest);
-                    item.Source = GetItemSourceCode(ingest, papername);
-                    item.Title = GetXmlData(story, Fields.Headline, ingest);
-                    item.Uid = GetXmlData(story, Fields.Id, ingest);
-                    item.Body = GetXmlData(story, Fields.Story, ingest);
-                    item.FilePath = document.Key;
-                    item.Summary = GetXmlData(story, Fields.Summary, ingest);
-                    item.Page = GetXmlData(story, Fields.Page, ingest);
-                    item.Section = GetXmlData(story, Fields.Section, ingest);
-
-                    item.Authors = GetAuthorList(GetXmlData(story, Fields.Author, ingest));
-                    item.PublishedOn = GetPublishedOn(GetXmlData(story, Fields.Date, ingest), ingest);
-
-                    articles.Add(item);
+                    var code = GetItemSourceCode(ingest, papername, sources);
+
+                    if (!string.IsNullOrEmpty(code)) // This is a valid newspaper source
+                    {
+                        var item = new SourceContent();
+                        item.Source = GetItemSourceCode(ingest, papername, sources);
+                        item.Title = GetXmlData(story, Fields.Headline, ingest);
+                        item.Uid = GetXmlData(story, Fields.Id, ingest);
+                        item.Body = GetXmlData(story, Fields.Story, ingest);
+                        item.FilePath = document.Key;
+                        item.Summary = GetXmlData(story, Fields.Summary, ingest);
+                        item.Page = GetXmlData(story, Fields.Page, ingest);
+                        item.Section = GetXmlData(story, Fields.Section, ingest);
+                        item.Language = ingest.GetConfigurationValue("language");
+                        item.ProductId = ingest.ProductId;
+
+                        item.Authors = GetAuthorList(GetXmlData(story, Fields.Author, ingest));
+                        item.PublishedOn = GetPublishedOn(GetXmlData(story, Fields.Date, ingest), ingest);
+
+                        articles.Add(item);
+                    }
                 }
             }
             catch (Exception ex)
```

Comment:
Doesn't seem right to log a warning and then throw the exception.

Reply:
I'm doing this because the exception that is thrown will not identify the name of the ingest being processed and that info might aid in troubleshooting. Do you think there's value in that? If not I can remove the try/catch block and let the exception bubble up.

Reply:
Then throw an exception with the appropriate information in it.  No need to log here, if it will be logged somewhere else.

Reply:
If I catch the exception that is generated by C#, then create and throw a new exception with a descriptive error, will I not lose the information associated with the original exception? This is why I log the warning, ensuring that information is saved, and then throw the original exception. I could also incorporate the original error in the message that I pass to my new exception so that both are reported when the new exception is thrown.

Reply:
You include the original in the new.  An exception has an innerException.

Reply:
Fixed it.